url28f181064f_simonjupp_java_skos_api_tgz-pJ8-example_CreateSKOSExampleJ8/src/uk/ac/manchester/cs/skos/SKOSDatasetImpl.java

```diff
@@ -0,0 +1,533 @@
+
+                SKOSEntity entOb = SKOSEntity.get(object.asOWLNamedIndividual().getIRI().toURI());
+                SKOSEntity entSub = SKOSEntity.get(subject.asOWLNamedIndividual().getIRI().toURI());
+
+                if (entOb == null) {
+                    entOb = new SKOSConceptImpl(object);
+                    SKOSEntity.put(object.asOWLNamedIndividual().getIRI().toURI(), entOb);
+                }
+                if (entSub == null) {
+                    entSub = new SKOSConceptImpl(subject);
+                    SKOSEntity.put(subject.asOWLNamedIndividual().getIRI().toURI(), entSub);
+                }
+
+                SKOSObjectProperty skosProp = skosManContent.getSKOSDataFactory().getSKOSObjectProperty(prop.asOWLObjectProperty().getIRI().toURI());
+                assertion.add(skosManContent.getSKOSDataFactory().getSKOSObjectRelationAssertion(entSub, skosProp, entOb));
+            }
+        }
+
+        return assertion;
+    }
+
+    public Set<SKOSDataRelationAssertion> getSKOSDataRelationAssertions(SKOSEntity skosEntity) {
+
+        Set<SKOSDataRelationAssertion> assertion = new HashSet<SKOSDataRelationAssertion>(50);
+
+        if (!SKOSEntity.containsKey(skosEntity.getURI())) {
+            SKOSEntity.put(skosEntity.getURI(), skosEntity);
+        }
+
+        for (OWLAxiom ax : owlOntology.getDataPropertyAssertionAxioms(skos2owlConverter.getAsOWLIndiviudal(skosEntity))) {
+            if (ax instanceof OWLDataPropertyAssertionAxiom) {
+                OWLDataPropertyAssertionAxiom  propAx = (OWLDataPropertyAssertionAxiom) ax;
+                OWLLiteral object = propAx.getObject();
+                OWLNamedIndividual subject = (OWLNamedIndividual) propAx.getSubject();
+                OWLDataPropertyExpression prop = propAx.getProperty();
+
+                SKOSEntity entSub = SKOSEntity.get(subject.asOWLNamedIndividual().getIRI().toURI());
+
+                if (entSub == null) {
+                    entSub = entityCreationHandler(subject);
+                    SKOSEntity.put(subject.asOWLNamedIndividual().getIRI().toURI(), entSub);
+                }
+                SKOSDataProperty skosProp = skosManContent.getSKOSDataFactory().getSKOSDataProperty(prop.asOWLDataProperty().getIRI().toURI());
+                SKOSLiteral literal = null;
+
+                if (!object.isRDFPlainLiteral()) {
+                    SKOSDataType type = skosManContent.getSKOSDataFactory().getSKOSDataType(object.getDatatype().getIRI().toURI());
+                    literal = skosManContent.getSKOSDataFactory().getSKOSTypedConstant(type, object.getLiteral());
+                }
+                else {
+                    literal = skosManContent.getSKOSDataFactory().getSKOSUntypedConstant(object.getLiteral(), object.getLang());
+                }
+
+                assertion.add(skosManContent.getSKOSDataFactory().getSKOSDataRelationAssertion(entSub, skosProp, literal));
+            }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 257:281, 314:338.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.